pkg/config/config_test.go

```diff
@@ -66,6 +66,43 @@ func TestFilterProperties(t *testing.T) {
 	assert.Check(t, properties["Captains"] != nil)
 }
 
+func TestFilterPropertiesToReturnExactMatchOnlyForConfigName(t *testing.T) {
+	m := make(map[string]map[string]string)
+
+	m["dashboard"] = make(map[string]string)
+	m["dashboard-availability"] = make(map[string]string)
+
+	properties := filterProperties("dashboard", m)
+
+	assert.Check(t, len(properties) == 1)
+	assert.Check(t, properties["dashboard"] != nil)
+}
+
+func TestFilterPropertiesToReturnExactMatchOnlyForConfigNameAndEnvironment(t *testing.T) {
+	m := make(map[string]map[string]string)
+
+	m["dashboard"] = make(map[string]string)
+	m["dashboard-availability"] = make(map[string]string)
+	m["dashboard.dev"] = make(map[string]string)
+	m["dashboard-availability.dev"] = make(map[string]string)
+
+	m["dashboard"]["prop1"] = "A"
+	m["dashboard"]["prop2"] = "A"
+	m["dashboard-availability"]["prop1"] = "A"
+	m["dashboard-availability"]["prop2"] = "A"
+
+	m["dashboard.dev"]["prop1"] = "B"
+	m["dashboard.dev"]["prop2"] = "C"
+	m["dashboard-availability.dev"]["prop1"] = "E"
+	m["dashboard-availability.dev"]["prop2"] = "F"
+
+	properties := filterProperties("dashboard.dev", m)
+
+	assert.Check(t, len(properties) == 1)
+	assert.Check(t, properties["dashboard.dev"] != nil)
+	assert.Check(t, len(properties["dashboard.dev"]) == 2)
+}
```

Comment:
Sorry about nitpicking, but these two assertions would make sense here:

```
assert.Equal(t, "B", properties["dashboard.dev"]["prop1"])
assert.Equal(t, "C", properties["dashboard.dev"]["prop2"])
```

Reply:
I thought about that but I deliberately chose against it. I can add them if really want, but I would rather create a separate test for this because if I add these assertions we're testing different things with one tiny unit test. The test name says it's only testing for exact match of config name, but the assertions you'd like to add here are testing a different thing. They're testing if more specific properties, if any, are overwriting more general ones. 

Or what do you think? I'll wait for your answer before adding / modifying the tests here.

Reply:
I see. Yes, you are completely right. By doing that, your test would be not tesing just a single thing anymore. Would be cool to have such a test with the assertions, since we are missing it currently.

Reply:
All right, I've implemented two more tests. Can you have a look? Thanks!

Reply:
Cool, thank you!